roles/moodle/templates/moodle-nginx.conf.j2

```diff
@@ -1,13 +1,21 @@
-location /moodle {
-    proxy_set_header X-Real-IP  $remote_addr;
-    proxy_set_header X-Forwarded-For $remote_addr;
-    proxy_set_header Host $host;
-    proxy_pass http://127.0.0.1:{{ apache_port }};
+location ~ ^/moodle/dataroot/ {
+    internal;
+    alias {{ moodle_data }};
 }
 
-location ~ ^/moodle.*\.php$ {
-    proxy_set_header X-Real-IP  $remote_addr;
-    proxy_set_header X-Forwarded-For $remote_addr;
-    proxy_set_header Host $host;
-    proxy_pass http://127.0.0.1:{{ apache_port }};
+location ~ ^/moodle(.*)\.php(.*)$ {
+    alias {{ moodle_base }}$1.php$2;
+    include fastcgi.conf;
+    fastcgi_split_path_info ^(.+\.php)(/.+)$;
+    fastcgi_read_timeout 300;
+    fastcgi_pass php;
+    fastcgi_index index.php;
+    fastcgi_param   PATH_INFO        $fastcgi_path_info;
+    fastcgi_param   SCRIPT_FILENAME   $document_root/$fastcgi_script_name;
+    fastcgi_param   PHP_VALUE "max_execution_time=300\n upload_max_filesize=500M\n post_max_size=500M\n max_input_vars=5000";
+}
+
+
+location ~ ^/moodle/ {
```

Comment:
This fix (appears) to solve the problem of http://box.lan/moodle 404's when the URL's trailing slash is missing.

```suggestion
location ~ ^/moodle {
```

Reply:
If restarting nginx doesn't puke, that seems reasonable

Reply:
This PR (but amended on Line 19 of moodle/templates/moodle-nginx.conf.j2) is being re-installed on a new VM 192.168.0.197 = 10.8.0.46 = 197-u2104-desk-moodle-nginx-0702

Using the same sub-minimal local_vars.yml ("Moodle only, nothing else") as yesterday.

To help evaluate the OSM theory and others as necessary.

Reply:
Conclusion:

http://10.8.0.46/moodle and http://10.8.0.46/moodle/ both work without OSM (on above NEW "Moodle only" VM).